src/Momento.Sdk/Internal/Retry/DefaultRetryEligibilityStrategy.cs

```diff
@@ -100,20 +125,26 @@ public bool IsEligibleForRetry<TRequest>(Status status, TRequest request)
 
         public override bool Equals(object obj)
         {
-            if ((obj == null) || !this.GetType().Equals(obj.GetType()))
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+            if (obj is null || obj.GetType() != GetType())
             {
                 return false;
             }
 
             var other = (DefaultRetryEligibilityStrategy)obj;
             return _retryableRequestTypes.SetEquals(other._retryableRequestTypes) &&
-                _retryableStatusCodes.SetEquals(other._retryableStatusCodes);
+                   _retryableStatusCodes.SetEquals(other._retryableStatusCodes);
         }
 
         public override int GetHashCode()
         {
-            return base.GetHashCode();
```

Comment:
We have several classes that use `base.GetHashCode()`, which violates the contract that two objects that equal each other should have the same hash code. They are mostly unimportant like this one, but we should update them when we see them.